json.go

```diff
@@ -19,23 +18,42 @@ func (db *DB) ExportCollection(collectionName string, exportPath string) error {
 	if !exists {
 		return ErrCollectionNotExist
 	}
-
-	result, err := db.FindAll(query.NewQuery(collectionName))
+	q := query.NewQuery(collectionName)
+	collectionCount, err := db.Count(q)
 	if err != nil {
 		return err
 	}
 
-	docs := make([]map[string]interface{}, 0)
-	for _, doc := range result {
-		docs = append(docs, doc.AsMap())
+	f, err := os.Create(exportPath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	if _, err = f.WriteString("["); err != nil {
+		return err
 	}
 
-	jsonString, err := json.Marshal(docs)
+	n := 0
+	err = db.ForEach(q, func(doc *d.Document) bool {
+		n++
+		jsonByte, err := json.Marshal(doc.AsMap())
+		if err != nil {
```

Comment:
I think it would be a good idea to recover this error in the outer scope.
If there is any failure inside this function, then the export process would not report an error, but the result would be clearly not correct.